datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -79,20 +97,36 @@ protected DurationDeserializer withLeniency(Boolean leniency) {
                     deser = deser.withLeniency(leniency);
                 }
             }
+            if (format.hasPattern()) {
+                deser = DurationUnitParser.from(format.getPattern())
+                        .map(deser::withPattern)
+                        .orElse(deser);
+            }
         }
         return deser;
     }
 
+    private DurationDeserializer withPattern(DurationUnitParser pattern) {
+        return new DurationDeserializer(this, pattern);
+    }
+
     @Override
     public Duration deserialize(JsonParser parser, DeserializationContext context) throws IOException
     {
         switch (parser.currentTokenId())
         {
             case JsonTokenId.ID_NUMBER_FLOAT:
-                BigDecimal value = parser.getDecimalValue();
-                return DecimalUtils.extractSecondsAndNanos(value, Duration::ofSeconds);
+                BigDecimal decValue = parser.getDecimalValue();
+                if (_durationUnitParser != null) {
+                    return _durationUnitParser.parse(decValue.longValue());
```

Comment:
This would be lossy... I wonder if the right handling would be throw an exception; or perhaps require a coercion check? (2.12 has `CoercionConfig` that could be checked to see if "floating-point -> integer" coercion is legal; by default it is, but can prevent.
This assuming that it is not possible to use fractions in meaningul way... is there?

Reply:
Limit the DurationParser to only `ID_NUMBER_INT` logic, to match the longs that are available in the `toXXX` methods of `Duration`, perhaps? (Now that I'm thinking about how this would be serialized.)
https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toDays--

Reply:
Yeah; it is easy enough to add support for `String` coercion if need be (if requested).
So let's go with that.

I might still go ahead and check for `CoercionConfig` for existing coercions -- I know "lenient" is used in some places, but I think my thinking has evolved so that there is difference between:

1. Coercion from different input value shape (like JSON String) into something that would usually be required (like integral number). For this we have CoercionConfig and a few legacy features. Leniency does not come to play as it is too coarse setting (on/off)
2. Leniency would affect handling within expected value domain -- so, for String representation of Date, whether we allow "2021-02-30", or similar (case-sensitivity option exists, too, which is yet separate)

Reply:
So leave this case block ( case JsonTokenId.ID_NUMBER_FLOAT:) as-is, with no changes, and the only case block that should be changed with this PR is the one for `JsonTokenId.ID_NUMBER_INT'? Keep things simpler for now?

Reply:
Well, kind of; change should be isolated in `_fromTimestamp()`, which can also be called from `_fromString()` -- for special case of XML, CSV etc (formats for which `ctxt.isEnabled(StreamReadCapability.UNTYPED_SCALARS)` returns true).
This because those formats can not reliably indicate number type (XML not without Schema, CSV not without something external), so auto-detection of integral numbers makes sense without coercion.

Reply:
Only integers are handled now